components/constellation/constellation.rs

```diff
@@ -800,6 +800,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn add_pending_change(&mut self, change: SessionHistoryChange)
+    {
+        self.handle_load_start_msg(change.new_pipeline_id);
+        self.pending_changes.push(change);
```

Comment:
I think I prefer the version which takes a `SessionHistoryChange` argument, since then the fields are named. But like you said, the important thing is that this always calls handle_load_start_msg.